Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraintPalmUp.cs

```diff
@@ -114,54 +157,168 @@ protected override bool IsValidController(IMixedRealityController controller)
-
-                            if (Vector3.Angle(palmPose.Up, handNormal) > flatHandThreshold)
+                            // If we are above the threshold angle, keep the menu mapped to the tracked object
+                            if (palmCameraAngle > followHandCameraFacingThresholdAngle)
+                            {
+                                RotationBehavior = SolverRotationBehavior.LookAtTrackedObject;
+                                OffsetBehavior = SolverOffsetBehavior.TrackedObjectRotation;
+                            }
+                            // If we are within the threshold angle, we snap to follow the camera
+                            else
                             {
-                                return false;
+                                RotationBehavior = SolverRotationBehavior.LookAtMainCamera;
+                                OffsetBehavior = SolverOffsetBehavior.LookAtCameraRotation;
                             }
                         }
-                    }
-
-                    // Check if the palm angle meets the prescribed threshold
-                    palmFacingThresholdMet = palmCameraAngle < facingCameraTrackingThreshold;
 
-                    // If using hybrid hand rotation, we proceed with additional checks
-                    if (followHandUntilFacingCamera && palmFacingThresholdMet)
-                    {
-                        // If we are above the threshold angle, keep the menu mapped to the tracked object
-                        if (palmCameraAngle > followHandCameraFacingThresholdAngle)
+                        if (useGazeActivation && (!eyeGazeActivationAlreadyTriggered || targetWorldLocked))
                         {
-                            RotationBehavior = SolverRotationBehavior.LookAtTrackedObject;
-                            OffsetBehavior = SolverOffsetBehavior.TrackedObjectRotation;
-                        }
-                        // If we are within the threshold angle, we snap to follow the camera
-                        else
-                        {
-                            RotationBehavior = SolverRotationBehavior.LookAtMainCamera;
-                            OffsetBehavior = SolverOffsetBehavior.LookAtCameraRotation;
+                            return IsUserGazingAtActivationPoint(jointedHand);
                         }
                     }
+
+                    eyeGazeActivationAlreadyTriggered = palmFacingThresholdMet ? eyeGazeActivationAlreadyTriggered : false;
+
+                    return palmFacingThresholdMet;
                 }
                 else
                 {
                     Debug.LogError("HandConstraintPalmUp requires controllers of type IMixedRealityHand to perform hand activation tests.");
                 }
 
+                eyeGazeActivationAlreadyTriggered = palmFacingThresholdMet ? eyeGazeActivationAlreadyTriggered : false;
```

Comment:
Looks like you are doign this line here, and above on line 187. Any reason for needing to do this twice?

Reply:
Good call out; I added it...out of desperation. It did nothing though, so removing it